.github/workflows/main.yaml

```diff
@@ -0,0 +1,73 @@
+name: Main CI
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.12'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+    - name: Install testing dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-test.txt
+    - name: Lint
+      # FIXME
+      continue-on-error: true
+      run: |
+        flake8 lib/pyld tests --count --show-source --statistics
+      env:
+        LOADER: ${{ matrix.loader }}
+  test:
+    runs-on: ubuntu-latest
+    needs: [lint]
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.8'
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+          - 'pypy3.10'
+        loader: [requests, aiohttp]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Install testing dependencies
+      run: |
+        pip install -r requirements-test.txt
+    - name: Fetch test suites
+      run: |
+        git clone --depth 1 https://github.com/w3c/json-ld-api.git _json-ld-api
+        git clone --depth 1 https://github.com/w3c/json-ld-framing.git _json-ld-framing
+        git clone --depth 1 https://github.com/json-ld/normalization.git _normalization
```

Comment:
These three aren't going to be necessary if https://github.com/digitalbazaar/pyld/pull/182 is merged.

Reply:
- I believe what I was thinking when this PR was written was to start with testing the code similar to how travis-ci was doing it but with github actions.
- This will give immediate familiar feedback for testing further changes.
- #182 is not addressing the test runner and code changes needed for rdf-canon tests.  Every rdf-canon test will be skipped leaving a large hole in coverage if we are to release any versions before the issue is addressed.  I don't think changing that test suite before rdf-canon is supported can be justified.
- It's arguably better to merge this *first* to get tests flowing.  #182 can be rebased off of it and will then have tests running.  The changes to make it use submodules are probably straightforward and would be best done with workflow tests to confirm.  (Assuming we can get outside contributor PR tests running properly.)

Reply:
> is not addressing the test runner and code changes needed for rdf-canon tests. Every rdf-canon test will be skipped

Should we fix that? I thought I [did add](https://github.com/digitalbazaar/pyld/pull/182/files#diff-fe7afb5c9c916e521401d3fcfb4277d5071798c3baf83baf11d6071742823584R9) that as a dependency and we should be running against that there...